docs/DeveloperGuide.md

```diff
@@ -2,37 +2,243 @@
+| Version | As a ... | I want to ...                              | So that I can ...                                             |
+|---------|----------|--------------------------------------------|---------------------------------------------------------------|
+| v1.0    | new user | see usage instructions                     | refer to them when I forget how to use the application        |
+| v1.0    | user     | add a recipe                               | curate my own unique recipes for future use                   |
+| v1.0    | user     | delete a recipe                            | remove any outdated or unimportant recipes                    |
+| v2.0    | user     | find a recipe by name                      | locate a recipe without having to go through the entire list  |
+| v2.0    | foodie   | save and load recipes                      | share recipes with friends conveniently                       |
+| v2.0    | user     | find recipes without a certain ingredient  | save time finding recipes that I can eat                      |
+| v2.0    | user     | find a recipe by date                      | find a recipe if I don't remember its name                    |
+
+### Use cases
+
+(For all use cases below, the **System** is the `RecipeIO` and the **Actor** is the `user`, unless specified otherwise)
+
+**Use case: Delete a recipe**
+
+1.  User requests to list recipes
+2.  RecipeIO shows a list of recipes
+3.  User requests to delete a specific recipe in the list
+4.  AddressBook deletes the person
+
+Use case ends.
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 
-## Glossary
+---
 
-* *glossary item* - Definition
+## Appendix: Commands Glossary
 
-## Instructions for manual testing
+* *add [NAME, MINUTES, KCALS, ALLERGIES, CATEGORY, URL]* - This is the command a user can call to add a recipe. 
+* *help* - This shows the user all the available commands. 
+* *list* - This lists out a users recipebook for them. 
+* *delete* - This deletes a recipe at a given valid index. If not valid, it will return an error message.
+* *find kw [KEYWORD]* - This finds recipes with a user-given keyword.
+* *find date [YYYY-MM-DD]* - This finds recipes added on a user-given date. The date is auto-added when adding. 
+* *find allergy [KEYWORD]* - This finds recipes without a certain ingredient.
+* *exit* - This is the command to leave the program. 
+
+---
+
+## **Appendix: Instructions for manual testing**
+
+Given below are instructions to test the app manually.
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** These instructions only provide a starting point for testers to work on;
+testers are expected to do more *exploratory* testing.</div>
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+### Adding a recipe
```

Comment:
The instruction for testing is detail, probably if a code snippet is attached for each example would be more straight forward.